src/main/scala/io/iohk/ethereum/blockchain/sync/regular/RegularSync.scala

```diff
@@ -26,11 +28,14 @@ class RegularSync(
 ) extends Actor
     with ActorLogging {
 
-  val fetcher: ActorRef =
-    context.actorOf(
-      BlockFetcher.props(peersClient, peerEventBus, self, syncConfig, blockValidator),
+  val fetcher: TypedActorRef[BlockFetcher.FetchCommand] =
+    context.spawn(
+      BlockFetcher(peersClient, peerEventBus, self, syncConfig, blockValidator),
       "block-fetcher"
     )
+
+  context.watch(fetcher)
```

Comment:
quite o few `watch` statements

maybe it would make sense to add:
```scala
      .receiveSignal {
        case (context, Terminated(ref)) =>
```